test/test_torch.py

```diff
@@ -8272,6 +8272,8 @@ def test_dim_reduction(self, device):
                  torch.int64,
                  torch.int32,
                  torch.int16]
+        if device.startswith('cuda'):  # 'cpu' and 'xla' do not support half
```

Comment:
`device.type == 'cuda'` is the more idiomatic way to test this

Reply:
but device is a string here